pkg/resource/repository/hook.go

```diff
@@ -0,0 +1,32 @@
+package repository
+
+import (
+	"context"
+
+	svcapitypes "github.com/aws-controllers-k8s/ecr-controller/apis/v1alpha1"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/ecr"
+)
+
+// addPutFieldsToSpec will describe the fields that are not return by
+// DescribeRepository calls
+func (rm *resourceManager) describeRepositoryFields(
```

Comment:
How about we call this `setResourceAdditionalFields` since it is setting fields on the resource.

Reply:
Agreed. Changed to `setResourceAdditionalFields`